sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs

```diff
@@ -0,0 +1,333 @@
+
+namespace Azure.ResourceManager.MigrationDiscoverySap.Tests.Tests
+{
+    public class MigrationSapDiscoveryTests : MigrationDiscoverySapManagementTestBase
+    {
+        public MigrationSapDiscoveryTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        public const string migrateProjectIdFormat = "/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Migrate/MigrateProjects/{2}";
+
+        [TestCase]
+        public async Task TestDiscoveryOperations()
+        {
+            SubscriptionResource subscription = await Client.GetDefaultSubscriptionAsync();
+
+            string rgName = Recording.GenerateAssetName("SdkTest-Net-");
+            string migrateProjName = Recording.GenerateAssetName("migrateProj");
+            string migrateProjectId = String.Format(migrateProjectIdFormat, subscription.Data.SubscriptionId, rgName, migrateProjName);
+            string discoverySiteName = Recording.GenerateAssetName("SapDiscoverySite-");
+
+            ResourceGroupResource rg = await CreateResourceGroup(subscription, rgName, AzureLocation.SoutheastAsia);
+
+            GenericResourceData data = new GenericResourceData(AzureLocation.SoutheastAsia)
+            {
+                Properties = new BinaryData(new Dictionary<string, object>
+                {
+                    { "registeredTools", new string[] { "ServerAssessment" }}
+                })
+            };
+
+            var operationStatus = await Client.GetGenericResources().CreateOrUpdateAsync(
+                WaitUntil.Completed, ResourceIdentifier.Parse(migrateProjectId), data);
+
+            // Create Discovery site payload
+            SAPDiscoverySiteData site = new SAPDiscoverySiteData();
+            site.MigrateProjectId = migrateProjectId;
+            site.Location = AzureLocation.SoutheastAsia;
+
+            // Create discovery site
+            ArmOperation<SAPDiscoverySiteResource> discoverySiteOp = await rg.GetSAPDiscoverySites().CreateOrUpdateAsync(
+                WaitUntil.Completed,
+                discoverySiteName,
+                site);
+            var resourceId = discoverySiteOp.Value.Id;
+
+            // Get SAP DiscoverySite
+            SAPDiscoverySiteResource sapDiscoverySiteResource = Client.GetSAPDiscoverySiteResource(resourceId);
+
+            // Post import entities
+            var importEntitiesOp = await sapDiscoverySiteResource.ImportEntitiesAsync(WaitUntil.Completed);
+
+            Assert.IsNotNull(
+                await TrackTillConditionReachedForAsyncOperationAsync(
+                    new Func<Task<bool>>(async () => await this.TrackForDiscoveryExcelInputSasUriAsync(
+                        importEntitiesOp)), 300),
+                "SAS Uri generation failed");
+
+            var opStatus = await importEntitiesOp.UpdateStatusAsync();
+            var map = JObject.Parse(opStatus.Content.ToString());
+            var inputExcelSasUri = map?["properties"]?["discoveryExcelSasUri"].ToString();
```

Comment:
can we just assign result to importEntitiesOp itself?

Reply:
Done